src/shared/codelens/codeLensUtils.ts

```diff
@@ -148,33 +150,27 @@ export function getMetricDatum({ command, isDebug, runtime }: {
 
 async function getAssociatedSamTemplate(
     documentUri: vscode.Uri,
-    workspaceFolderUri: vscode.Uri
+    workspaceFolderUri: vscode.Uri,
+    handlerName: string
 ): Promise<vscode.Uri> {
-
-    // Get Template files in Workspace
-    const templatesAsync: AsyncIterableIterator<vscode.Uri> = detectLocalTemplates({
+    const templates = detectLocalTemplates({
         workspaceUris: [workspaceFolderUri]
     })
 
-    // See which templates (if any) are in a folder, or parent folder, of the document of interest
-    const templates = await toArrayAsync(templatesAsync)
-    const candidateTemplates = templates
-        .filter(template => {
-            const folder = path.dirname(template.fsPath)
-
-            return documentUri.fsPath.indexOf(folder) === 0
-        })
+    for await (const template of templates) {
+        try {
+            // Throws if template does not contain a resource for this handler.
+            await CloudFormation.getResourceFromTemplate({
+                templatePath: template.fsPath,
+                handlerName
+            })
+        } catch {
+            continue
```

Comment:
Nit: Haven't seen any other `continue`s in the codebase...should we `return` in the `try` and have an empty `catch`?

Reply:
I think this is fine. We could move the return inside of the `try`, but you need a `catch` (or `finally`) block anyway, so it doesn't result in cleaner code IMO.